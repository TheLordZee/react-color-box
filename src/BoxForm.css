.BoxForm {
    display: flex;
    justify-content: center;
}

.BoxForm-input {
    margin-right: 5vw;
}
