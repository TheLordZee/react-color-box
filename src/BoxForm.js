import "./BoxForm.css"

const BoxForm = ({handleSubmit, onChange, formData}) => (
    <form className="BoxForm" onSubmit={handleSubmit}>
        <label htmlFor="height">Height:</label>
        <input 
            id="height" 
            className="BoxForm-input"
            name="height" 
            type="number" 
            onChange={onChange} 
            value={formData.height}
        />

        <label htmlFor="width">Width:</label>
        <input 
            id="width" 
            className="BoxForm-input"
            name="width" 
            type="number" 
            onChange={onChange}
            value={formData.width}
        />

        <label htmlFor="color">Color:</label>
        <input 
            id="color" 
            className="BoxForm-input"
            name="color" 
            type="text"
            onChange={onChange}
            value={formData.color}
        />
        <br/>
        <button className="BoxForm-btn">Add Box</button>
    </form>
)

export default BoxForm;