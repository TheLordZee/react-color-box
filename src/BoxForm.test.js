import { render, screen } from '@testing-library/react';
import BoxForm from "./BoxForm"
const data = {
    height: "",
    width: "",
    color: ""
}
it('renders', () => {
    render(<BoxForm formData={data}/>);
});


it("matches snapshot", () => {
    const {asFragment} = render(<BoxForm formData={data}/>);
    expect(asFragment()).toMatchSnapshot();
})