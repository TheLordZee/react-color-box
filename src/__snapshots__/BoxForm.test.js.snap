// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="BoxForm"
  >
    <label
      for="height"
    >
      Height:
    </label>
    <input
      class="BoxForm-input"
      id="height"
      name="height"
      type="number"
      value=""
    />
    <label
      for="width"
    >
      Width:
    </label>
    <input
      class="BoxForm-input"
      id="width"
      name="width"
      type="number"
      value=""
    />
    <label
      for="color"
    >
      Color:
    </label>
    <input
      class="BoxForm-input"
      id="color"
      name="color"
      type="text"
      value=""
    />
    <br />
    <button
      class="BoxForm-btn"
    >
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
